File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/CleaningMap.java
package com.neatorobotics.sdk.android.models;

import java.io.Serializable;
import java.util.Date;

public class CleaningMap implements Serializable {

    private String url, id, startAt, endAt, error;
    private boolean docked, delocalized;
    private double area, chargingTimeSeconds, timeInError, timeInPause;
    private Date urlExpiresAt;

    public boolean isExpired() {
        //milliseconds
        long difference = urlExpiresAt.getTime() - System.currentTimeMillis();
        if( difference < 30*1000 ) return true; // 30 secs of margin
        else return false;
    }

    public boolean isDelocalized() {
        return delocalized;
    }

    public void setDelocalized(boolean delocalized) {
        this.delocalized = delocalized;
    }

    public Date getUrlExpiresAt() {
        return urlExpiresAt;
    }

    public void setUrlExpiresAt(Date urlExpiresAt) {
        this.urlExpiresAt = urlExpiresAt;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getStartAt() {
        return startAt;
    }

    public void setStartAt(String startAt) {
        this.startAt = startAt;
    }

    public String getEndAt() {
        return endAt;
    }

    public void setEndAt(String endAt) {
        this.endAt = endAt;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public boolean isDocked() {
        return docked;
    }

    public void setDocked(boolean docked) {
        this.docked = docked;
    }

    public double getArea() {
        return area;
    }

    public void setArea(double area) {
        this.area = area;
    }

    public double getChargingTimeSeconds() {
        return chargingTimeSeconds;
    }

    public void setChargingTimeSeconds(double chargingTimeSeconds) {
        this.chargingTimeSeconds = chargingTimeSeconds;
    }

    public double getTimeInError() {
        return timeInError;
    }

    public void setTimeInError(double timeInError) {
        this.timeInError = timeInError;
    }

    public double getTimeInPause() {
        return timeInPause;
    }

    public void setTimeInPause(double timeInPause) {
        this.timeInPause = timeInPause;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/CleaningMap.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.io.Serializable
import java.util.Date

@Parcelize
class CleaningMap(
    var url: String? = null,
    var id: String? = null,
    var startAt: String? = null,
    var endAt: String? = null,
    var error: String? = null,
    var status: String? = null,
    var isDocked: Boolean = false,
    var isDelocalized: Boolean = false,
    var isValidAsPersistentMap: Boolean = false,
    var area: Double = 0.toDouble(),
    var chargingTimeSeconds: Double = 0.toDouble(),
    var timeInError: Double = 0.toDouble(),
    var timeInPause: Double = 0.toDouble(),
    var urlExpiresAt: Date? = null,
    var rotation:Int = 0
) : Parcelable {

    //milliseconds
    //30 secs of margin
    val isExpired: Boolean
        get() {
            val difference = urlExpiresAt!!.time - System.currentTimeMillis()
            return difference < 30 * 1000
        }

    companion object {

        val COMPLETE = "complete"
        val INCOMPLETE = "incomplete"
        val CANCELLED = "cancelled"
        val INVALID = "invalid"
    }
}
